package.json

```diff
@@ -32,7 +32,9 @@
     "ubiquity",
     "open-source"
   ],
-  "dependencies": {},
+  "dependencies": {
+    "@supabase/supabase-js": "^2.45.2"
```

Comment:
I think this should be made optional. It's a heavy library with lots of dependencies that will not be required in many projects but will be installed regardless.

Reply:
So we place it into `devDependencies` so it's not added to the build, then any projects it's used they'll need to install `supabase` separately? So that's something that users will need to be mindful of.

So why don't I change it so that the logger takes in `SupabaseClient` vs `key & url` that way there can be no confusion about things?

Reply:
Thinking in terms of plugins, we need to create a `supabaseClient` to init the `adapters` anyway so we can pass that into the logger at the same time so client vs vars seems cleaner and it's the only way that we can guarantee from a package perspective that the invoker has installed `supabase-js`.

Reply:
I think this could go to `peerDependencies` or `optionalDependencies`. My thinking is that a project that doesn't want to use Supabase doesn't need it to be installed. It could also be a separate project, or be abstracted with an interface.

Reply:
I'm not sure of the difference with those options. I know `devDeps` excludes it from the build and will never attempt to introduce it into a project as it relies on the invoker having installed it either as a primary or a devDep itself.

I think `devDeps` covers your use case but I will have to look into what the differences are unless you know which one you'd prefer?

Reply:
Easiest to use is always the best!

Reply:
I'm not expert at handling dependencies but I think that it doesn't matter where you put the dependency because the file is trying to import Supabase. Unless there is a way to do it with dynamic import?

Reply:
Maybe peer dependencies is a better approach than `devDeps` only if it does not enforce that the package is installed? `optionalDeps` appears to still install it according to gpt4 higher up the convo, it was struggling to grasp that we do not want to enforce an install.

![image](https://github.com/user-attachments/assets/d6fab84e-da0f-454a-adc0-8835f7028a7e)

Reply:
peerDeps are installed by npm, but not by yarn. Packages can be dynamically imported. I did that once but I can't find the project.